### PR TITLE
ADT: Remove Azure.Core project reference to use shipped package

### PR DESCRIPTION
**Pull request by drwill-ms**
Pavel already removed the project reference for us, but we can remove the .csproj from our solution.

https://github.com/Azure/azure-sdk-for-net/pull/16387/files

### CODE REVIEW

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/DigitalTwinsClientSample.csproj [hunk 1]
```diff
@@ -8,10 +8,10 @@
 
   <ItemGroup>
     <!--using the preview version of Azure.Identity to have access to KnownAuthorityHosts static class.-->
-    <PackageReference Include="Azure.Identity" Version="1.2.0-preview.2" />
+    <PackageReference Include="Azure.Identity" Version="1.2.1" />
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/DigitalTwinsClientSample.csproj:11]: Bring these inline with the versions of our other packages

File: sdk/digitaltwins/Azure.DigitalTwins.Core/Azure.DigitalTwins.Core.sln [hunk 2]
```diff
@@ -21,8 +21,6 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Azure.Core.TestFramework", "..\..\core\Azure.Core.TestFramework\src\Azure.Core.TestFramework.csproj", "{1FC8A3EA-3C0D-4DDF-B710-A7091F2CEBB1}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Azure.Core", "..\..\core\Azure.Core\src\Azure.Core.csproj", "{861E603A-6ADF-41C2-9364-C39499C08F25}"
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/Azure.DigitalTwins.Core.sln:24]: Pavel shipped Azure.Core package, so we can revert to nuget reference.

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/DigitalTwinsClientSample.csproj [hunk 3]
```diff
@@ -8,10 +8,10 @@
...
     <PackageReference Include="CommandLineParser" Version="2.7.82" />
 
-    <PackageReference Include="System.Text.Json" Version="4.7.1" />
+    <PackageReference Include="System.Text.Json" Version="4.6.0" />
```
Comment (vinagesh) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/DigitalTwinsClientSample.csproj:14]: Downgrading?

**Review by vinagesh — APPROVED** [review]

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/DigitalTwinsClientSample.csproj (hunk 3 above)
Comment (drwill-ms) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/DigitalTwinsClientSample.csproj:14]: To match the other projects.

**Review by timtay-microsoft — APPROVED** [review]
Looks like Pavel already did some of the changes in our .csproj. With that in mind, this looks good to me

**Pull request merged**